homeassistant/components/duwi/__init__.py

```diff
@@ -0,0 +1,380 @@
+"""Support for Duwi Smart devices."""
+
+import asyncio
+import functools
+import time
+from typing import Any, NamedTuple
+
+from duwi_smarthome_sdk.base_api import CustomerApi, SharingTokenListener
+from duwi_smarthome_sdk.device_scene_models import CustomerDevice, CustomerScene
+from duwi_smarthome_sdk.manager import Manager, SharingDeviceListener
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import __version__
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import ConfigEntryAuthFailed
+from homeassistant.helpers import config_validation as cv, device_registry as dr
+from homeassistant.helpers.dispatcher import dispatcher_send
+from homeassistant.helpers.typing import ConfigType
+
+from .const import (
+    _LOGGER,
+    ACCESS_TOKEN,
+    ADDRESS,
+    APP_KEY,
+    APP_SECRET,
+    CLIENT_MODEL,
+    CLIENT_VERSION,
+    DOMAIN,
+    DUWI_DISCOVERY_NEW,
+    DUWI_HA_SIGNAL_UPDATE_ENTITY,
+    DUWI_SCENE_UPDATE,
+    HOUSE_KEY,
+    HOUSE_NAME,
+    HOUSE_NO,
+    PASSWORD,
+    PHONE,
+    REFRESH_TOKEN,
+    SUPPORTED_PLATFORMS,
+    WS_ADDRESS,
+)
+
+type DuwiConfigEntry = ConfigEntry[HomeAssistantDuwiData]
+
+CONFIG_SCHEMA = cv.config_entry_only_config_schema(domain=DOMAIN)
+
+
+class HomeAssistantDuwiData(NamedTuple):
+    """Duwi data stored in the Home Assistant data object."""
+
+    manager: Manager
+    listener: SharingDeviceListener
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up the Duwi Smart Devices integration."""
+
+    # Check for existing config entries for this integration
+    hass.data.setdefault(DOMAIN, {})
+    if not hass.config_entries.async_entries(DOMAIN):
+        # No entries found, initiate the configuration flow
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": config_entries.SOURCE_IMPORT}, data={}
+            )
+        )
+
+    # Setup was successful
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: DuwiConfigEntry) -> bool:
+    """Set up a config entry for the Duwi Smart Devices integration."""
+
+    hass.data.setdefault(DOMAIN, {})
+
+    token_listener = TokenListener(hass, entry)
+    manager: Manager = Manager(
+        _id=entry.entry_id,
```

Comment:
why do we pass the id?

Reply:
Thank you for the reminder. I will no longer pass this parameter in